### DISCUSSION TITLE
Importing Data - Default vs Custom

### DISCUSSION THREAD

**Discussion by Barrow1990** [firefly-iii/firefly-iii#9260] [Firefly III]
Hi all.
So currently looking at how im going to use Firefly III and seem to have a reasonable understanding how i want to do things.
Now im thinking long term importing.

So currently i have CSV file from my bank which has 4 lines that requires removing at the start due to it not being in the correct CSV format.

This then results in the CSV to be properly formed with the headers at top.
My bank in question is Nationwide. Which outputs in the form of:
| Date | Transaction Type | Description | Paid out | Paid in | Balance |
| --- | --- | --- | --- | --- | --- |
| 22-Sept-24 | Bank credit PLACE OF WORK | Bank credit PLACE OF WORK | | £2000 | £2000 |
| 22-Sept-24 | Contactless Payment | TESCO STORES 1234 | £200 | | £1800 |

Note: The options that come up in the Transaction Type are:
Contactless Payment, Visa Purchase, Transfer from, Direct debit

Is it better for me to categorise outside and then just import via api with the relevant fields or is it best practice to import the csv as raw as possible and let firefly do all the stuff with rules?

**Comment by JC5** [comment]
I see most people using rules to manage this. It's not "better" either way though, because rules have their limits too.

**Reply by Barrow1990** [reply]
So it will be a case of which ever method the user prefers. I feel long term rules will be better... but it will get complicated on how  the rules need to be created and with the quantity. I think for me it will be better to do the modifying in Python then import via api
